tensorboard/components/tf_backend/test/backendTests.ts

```diff
@@ -115,68 +117,69 @@ describe('backend tests', () => {
 
       describe('#pluginRoute', () => {
         it('encodes slash correctly', function() {
-          assert.equal(
+          assertRelativePath(
               this.router.pluginRoute('scalars', '/scalar'),
-              '/data/plugin/scalars/scalar');
+              'data/plugin/scalars/scalar');
         });
 
         it('encodes query param correctly', function() {
-          assert.equal(
+          assertRelativePath(
               this.router.pluginRoute(
                   'scalars',
                   '/a',
                   createSearchParam({b: 'c', d: ['1', '2']})),
-              '/data/plugin/scalars/a?b=c&d=1&d=2');
+              'data/plugin/scalars/a?b=c&d=1&d=2');
         });
 
         it('encodes parenthesis correctly', function() {
-          assert.equal(
+          assertRelativePath(
               this.router.pluginRoute('scalars', '/a',
               createSearchParam({foo: '()'})),
-              '/data/plugin/scalars/a?foo=%28%29');
+              'data/plugin/scalars/a?foo=%28%29');
         });
 
         it('encodes query param the same as #addParams', function() {
-          assert.equal(
+          assertRelativePath(
               this.router.pluginRoute(
                   'scalars',
                   '/a',
                   createSearchParam({b: 'c', d: ['1']})),
-              addParams('/data/plugin/scalars/a', {b: 'c', d: ['1']}));
-          assert.equal(
+              addParams('data/plugin/scalars/a', {b: 'c', d: ['1']}));
+          assertRelativePath(
               this.router.pluginRoute(
                   'scalars',
                   '/a',
                   createSearchParam({foo: '()'})),
-              addParams('/data/plugin/scalars/a', {foo: '()'}));
+              addParams('data/plugin/scalars/a', {foo: '()'}));
         });
 
         it('ignores custom extension', function() {
-          assert.equal(
+          assertRelativePath(
               this.router.pluginRoute('scalars', '/a', undefined, 'meow'),
```

Comment:
Why does this typecheck? The `Router` interface declares `pluginRoute`
as a function taking two parameters, not four. Do we have implicit `any`
propagation somehow?

Reply:
Thanks for catching the error!

Apparently this is legal 🤷‍♂️ :  http://www.typescriptlang.org/play/#src=export%20interface%20Router%20%7B%0D%0A%20%20%20%20foo%3A%20(bar%3A%20boolean%2C%20baz%3A%20string)%20%3D%3E%20string%3B%0D%0A%7D%3B%0D%0A%0D%0Aexport%20function%20createRouter()%3A%20Router%20%7B%0D%0A%20%20%20%20return%20%7B%0D%0A%20%20%20%20%20%20%20%20foo%3A%20(bar%3A%20boolean%2C%20baz%3A%20string%2C%20random%3F%3A%20string)%3A%20string%20%3D%3E%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20return%20'hello'%3B%0D%0A%20%20%20%20%20%20%20%20%7D%2C%0D%0A%20%20%20%20%7D%3B%0D%0A%7D

Fixed it.

Reply:
No, that’s not what I’m concerned about. That’s perfectly fine.

At the _callsite_ in this comment, you are invoking a function with more
arguments than the number of declared parameters. [TypeScript disallows
this][1] (which it must—this is the only sound thing to do). But this
type error was not caught. Why?

[1]: http://www.typescriptlang.org/play/#src=export%20interface%20Router%20%7B%0D%0A%20%20%20%20foo%3A%20(bar%3A%20boolean%2C%20baz%3A%20string)%20%3D%3E%20string%3B%0D%0A%7D%3B%0D%0A%0D%0A%2F%2F%20This%20typechecks%20(good).%0D%0Aexport%20function%20createRouter()%3A%20Router%20%7B%0D%0A%20%20%20%20return%20%7B%0D%0A%20%20%20%20%20%20%20%20foo%3A%20(bar%3A%20boolean%2C%20baz%3A%20string%2C%20random%3F%3A%20string)%3A%20string%20%3D%3E%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20return%20'hello'%3B%0D%0A%20%20%20%20%20%20%20%20%7D%2C%0D%0A%20%20%20%20%7D%3B%0D%0A%7D%0D%0A%0D%0A%2F%2F%20This%20errors%20(good).%20Why%20does%20it%20not%20error%20in%20your%20%60assertRelativePath%60%3F%0D%0AcreateRouter().foo(true%2C%20%22baz%22%2C%20666)%3B%0D%0A